jmh-benchmarks/src/main/java/org/apache/kafka/jmh/group_coordinator/ServerSideAssignorBenchmark.java

```diff
@@ -0,0 +1,185 @@
+        }
+
+        if (isReassignment) {
+            GroupAssignment initialAssignment = partitionAssignor.assign(assignmentSpec, subscribedTopicDescriber);
+            Map<String, MemberAssignment> members;
+
+            members = initialAssignment.members();
+
+            // Update the AssignmentSpec with the results from the initial assignment.
+            Map<String, AssignmentMemberSpec> updatedMembers = new HashMap<>();
+
+            members.forEach((memberId, memberAssignment) -> {
+                AssignmentMemberSpec memberSpec = assignmentSpec.members().get(memberId);
+                updatedMembers.put(memberId, new AssignmentMemberSpec(
+                    memberSpec.instanceId(),
+                    memberSpec.rackId(),
+                    memberSpec.subscribedTopicIds(),
+                    memberAssignment.targetPartitions()
+                ));
+            });
+
+            // Add new member to trigger a reassignment.
+            Optional<String> rackId = isRackAware ? Optional.of("rack" + (memberCount + 1) % numberOfRacks) : Optional.empty();
+
+            updatedMembers.put("newMember", new AssignmentMemberSpec(
+                Optional.empty(),
+                rackId,
+                topicMetadata.keySet(),
+                Collections.emptyMap()
+            ));
+
+            this.assignmentSpec = new AssignmentSpec(updatedMembers);
+        }
+    }
+
+    private Map<Integer, Set<String>> mkMapOfPartitionRacks(int numPartitions) {
+        Map<Integer, Set<String>> partitionRacks = new HashMap<>(numPartitions);
+        for (int i = 0; i < numPartitions; i++) {
+            partitionRacks.put(i, new HashSet<>(Arrays.asList("rack" + i % numberOfRacks, "rack" + (i + 1) % numberOfRacks)));
+        }
+        return partitionRacks;
+    }
+
+    private void addTopicSubscriptions(Map<Uuid, TopicMetadata> topicMetadata) {
+        Map<String, AssignmentMemberSpec> members = new TreeMap<>();
+        List<Uuid> allTopicIds = new ArrayList<>(topicMetadata.keySet());
+        int topicCounter = 0;
+
+        for (int i = 0; i < memberCount; i++) {
+            String memberName = "member" + i;
+            Optional<String> rackId = isRackAware ? Optional.of("rack" + i % numberOfRacks) : Optional.empty();
+            List<Uuid> subscribedTopicIds;
+
+            // When subscriptions are uniform, all members are assigned all topics.
+            if (isSubscriptionUniform) {
+                subscribedTopicIds = allTopicIds;
+            } else {
+                subscribedTopicIds = Arrays.asList(
+                    allTopicIds.get(i % topicCount),
+                    allTopicIds.get((i+1) % topicCount)
+                );
+                topicCounter = max (topicCounter, ((i+1) % topicCount));
+
+                if (i == memberCount - 1 && topicCounter < topicCount - 1) {
+                    subscribedTopicIds.addAll(allTopicIds.subList(topicCounter + 1, topicCount - 1));
+                }
```

Comment:
I don't fully understand you selection logic. Could you elaborate?

Reply:
We assign every member two different topics and if there's any remaining topics assign them all to the last member
I tried to think of many ways to distribute them better but it get's too complicated and prone to error based on the input numbers. If you have any other ideas lmk

Reply:
Yeah I thought of a couple of ways but they all require restrictions on the input, Another way is to round robin the last remaining topics amongst all the members OR

Reply:
I like the bucket approach that I mentioned previously. Don't you think that it is a more realistic distribution?

Reply:
yeah makes sense, I implemented it, but my only worry is that the max topics any member can get will be topicCount/5